app/questionnaire/questionnaire_schema.py

```diff
@@ -302,6 +303,9 @@ def _get_blocks_by_id(self) -> dict[str, ImmutableDict]:
                     "PrimaryPersonListCollector",
                     "RelationshipCollector",
                 ):
+                    self._list_name_to_section_id_origin_map[
+                        block["for_list"]
+                    ] = self._parent_id_map[group["id"]]
```

Comment:
Im not sure if the naming is accurate. This wouldn't be the origin, it would be the last section to use this list name. If 3 sections used the same list name, the id would be for section 3.

I don't think we have any validation to prevent a list name being used by multiple list collectors as long as they are in separate sections?. Therefore, this would need to hold a list of section id, then you need to loop over each list collector for each section and handle dependencies.

Reply:
Related to https://github.com/ONSdigital/eq-questionnaire-validator/pull/205#discussion_r1273889828

Reply:
Updated to handle multiple list collectors in separate sections as discussed, in-line with other PR referenced above. A new ticket will be created to add tests for the scenarios where we have multiple list collectors.